### PR TITLE
Handle files and spans that don't exist

### LINKED ISSUE #37: ENOENT error when using `psa` to compile
See https://github.com/purescript-contrib/pulp/issues/354; this seems to me to be a psa issue rather than a pulp issue. Have you seen this before / can you help? I have seen this locally with pulp 12.3.0 and psa 0.7.2 but I haven't been able to reproduce it consistently.

### PR DESCRIPTION
**Pull request by natefaubion**
Fixes #37

### CODE REVIEW

File: src/Main.purs [hunk 1]
```diff
@@ -216,18 +216,25 @@ main = void do
     | otherwise =
        throwException (Child.toStandardError err)
 
+  isEmptySpan filename pos =
+    filename == "" ||
```
Comment (garyb) [inline src/Main.purs]: Another filename case here would be `"<module>"`, although I think the pos guarding will cover the cases that arises in too.

Comment (natefaubion) [reply src/Main.purs]: Why would "<module>" be a file name?

Comment (garyb) [reply src/Main.purs]: https://github.com/purescript/purescript/issues/3417

Comment (garyb) [reply src/Main.purs]: As for actually why, I don't know, it's undoubtedly something stupid I did though :smile:

Comment (natefaubion) [reply src/Main.purs]: There are too many possible cases to count and guard against 😭 :rage4:

Comment (garyb) [reply src/Main.purs]: Yeah - it is the compiler's fault anyway. Especially given sometimes these error message have the module information, but still lack a span.

**Pull request merged**